test/src/ui/widget.spec.ts

```diff
@@ -0,0 +1,1792 @@
+        parent.layout = layout;
+        widget.parent = parent;
+        widget.parent = null;
+        expect(layout.methods.indexOf('onChildRemoved')).to.not.be(-1);
+      });
+
+    });
+
+    describe('#dispose()', () => {
+
+      it('should dispose of the resource held by the widget', () => {
+        let widget = new Widget();
+        let layout = new LogLayout();
+        widget.layout = layout;
+        layout.dispose();
+        expect(layout.parent).to.be(null);
+      });
+
+      it('should be called automatically when the parent is disposed', () => {
+        let widget = new Widget();
+        let layout = new LogLayout();
+        widget.layout = layout;
+        widget.dispose();
+        expect(layout.parent).to.be(null);
+        expect(layout.isDisposed).to.be(true);
+      });
+
+    });
+
+    describe('#isDisposed', () => {
+
+      it('should test whether the layout is disposed', () => {
+        let layout = new LogLayout();
+        expect(layout.isDisposed).to.be(false);
+        layout.dispose();
+        expect(layout.isDisposed).to.be(true);
+      });
+
+      it('should be read-only', () => {
+        let layout = new LogLayout();
+        expect(() => { layout.isDisposed = false; }).to.throwError();
+      });
+
+    });
+
+    describe('#parent', () => {
+
+      it('should get the parent widget of the layout', () => {
+        let layout = new LogLayout();
+        let parent = new Widget();
+        parent.layout = layout;
+        expect(layout.parent).to.be(parent);
+      });
+
+      it('should throw an error if set to `null`', () => {
+        let layout = new LogLayout();
+        expect(() => { layout.parent = null; }).to.throwError();
+      });
+
+      it ('should throw an error if set to a different value', () => {
+        let layout = new LogLayout();
+        let parent = new Widget();
+        parent.layout = layout;
+        expect(() => { layout.parent = new Widget(); }).to.throwError();
+      });
+
+      it('should be a no-op if the parent is set to the same value', () => {
+        let layout = new LogLayout();
+        let parent = new Widget();
+        parent.layout = layout;
+        layout.parent = parent;
+        expect(layout.parent).to.be(parent);
+      });
+
+    });
+
+    describe('#onResize()', () => {
+
+      it("should be invoked on a `resize` message", () => {
+        let layout = new LogLayout();
+        let parent = new Widget();
+        parent.layout = layout;
+        sendMessage(parent, ResizeMessage.UnknownSize);
+        expect(layout.methods.indexOf('onResize')).to.not.be(-1);
+      });
+
+      it('should send a `resize` message to each of the widgets in the layout', () => {
+        let layout = new LogLayout();
+        let parent = new Widget();
+        parent.layout = layout;
+        sendMessage(parent, ResizeMessage.UnknownSize);
+        expect(layout.methods.indexOf('onResize')).to.not.be(-1);
+        let child = layout.iter().next();
```

Comment:
you should test each child, not just the first. The `iteration` module has helpers for this:

`expect(every(layout, child => child.methods.indexOf('onResize') !== -1)).to.be(true)`